### DISCUSSION TITLE
How to use instantsearch without searchbox but autocomplete? (nextjs)

### DISCUSSION THREAD

**Discussion by mastoj** [algolia/instantsearch#5346] [Q&A]
I want to use instantsearch with algolia in a nextjs app, but struggle to get it to play along. The use case I have is to have an autocomplete searchbox in the header on all pages and then a dedicated search page. When you click or select a search result in the autocomplete box you should be redirected to the search page, but I can't get that to play nicely when the autocomplete box isn't inside the same `InstantSearch` provider.

I'm using nextjs 13 and also the new app folder, but it should still be doable.

So conceptually I have a `layout.tsx` where the autocomplete is then I have a `search/page.tsx` where I have defined my `InstantSearch` provider and shows the search results. If there is a good way to trigger a search without a searchbox inside the `InstantSearch` provider I think I would be good.

**Comment by Haroenv** [comment]
@mastoj which autocomplete is that? do you have a sandbox with the existing code so we can help you?

**Reply by mastoj** [reply]
I created a small example that doesn't work as I want it to, but might convey what I'm after: https://codesandbox.io/p/sandbox/nextjs-algolia-target-jhxy2r

The search box is an autocomplete box, and when you hit enter I want it to take you to the search result page with the search result for what you enter in the box. You can ignore the autocomplete suggestions, that is irrelevant but there to show the target end state. So the goal is, when hitting enter in for your search term you should be taken to the search page showing the results there, and when on the search page and you type a new search term and hit enter it should of course be updated.

Another thing that isn't optimal is how the actual input field is loaded. Since the input field is injected server side it pops out instead of being there in the initial render which is sort of annoying.

**Reply by Haroenv** [reply]
If I understand what you're trying to do: 

https://codesandbox.io/p/sandbox/nextjs-algolia-target-forked-loe7ik?file=%2Fapp%2FAutocomplete.tsx&selection=%5B%7B%22endColumn%22%3A9%2C%22endLineNumber%22%3A46%2C%22startColumn%22%3A1%2C%22startLineNumber%22%3A43%7D%5D for setting the query in the autocomplete from the URL

https://codesandbox.io/p/sandbox/nextjs-algolia-target-forked-loe7ik?file=%2Fapp%2Fsearch%2Fpage.tsx&selection=%5B%7B%22endColumn%22%3A1%2C%22endLineNumber%22%3A51%2C%22startColumn%22%3A1%2C%22startLineNumber%22%3A50%7D%5D for ensuring query is taken in account

For using autocomplete with server side rendering, you should use the react renderer (a little more code) as presented here: https://github.com/algolia/autocomplete/tree/next/examples/react-renderer

**Reply by mastoj** [reply]
oh, I can actually do server side rendering with that one... that might be interesting.

With that said, both your samples fail. When hitting enter the url is updated and then the query is set to empty right after. I also added my own interpretation of a virtual searchbox, but yours was nice as well.

I think I have the puzzle pieces I need now and can move forward. Thanks.

**Reply by Haroenv** [reply]
> and then the query is set to empty right after

I think that may be a bug with codesandbox, as I didn't see that when opening the example in a new tab.

**Reply by mastoj** [reply]
I can't get it to work 100%. Changing the url directly, but if you hit enter in the search box and push a new url the result doesn't update even though the url was updated.
